.github/workflows/test.yaml

```diff
@@ -0,0 +1,83 @@
+on: [push, pull_request]
+name: Test
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.14.x
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Lint Dependencies
+        run: make build-deps
+      - name: Lint
+        run: |
+          export PATH=$PATH:$(go env GOPATH)/bin
+          make lint
+      - name: Spellcheck
+        run: |
+          export PATH=$PATH:$(go env GOPATH)/bin
+          make spellcheck
+  test:
+    strategy:
+      matrix:
+        go-version: [1.14.x]
+        platform: [ubuntu-latest, macos-latest, ubuntu-16.04]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Test
+        run: make test
+
+  cover:
+    strategy:
+      matrix:
+        go-version: [1.14.x]
+        platform: [ubuntu-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Test
+        run: make cover
+      - name: CodeCov
+        uses: codecov/codecov-action@v1.0.4
+        with:
+          # The token below is used exclusively for uploading coverage reports.
+          token: "d82eb1d7-5990-4a31-baa4-156473402105"
+          file: ./coverage.txt
+
+  build:
+    strategy:
+      matrix:
+        go-version: [1.14.x]
+        platform: [ubuntu-latest, macos-latest, ubuntu-16.04]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: build
+        run: make build
+
+  build-docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: build
+        run: docker build .
```

Comment:
Do we want this step to upload or push the compile docker image / push them?

Reply:
Not yet.  Just validating the build here.